From cc21b9b7c442098989d8a9b64cf8adb5efb1ebd5 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Thu, 14 Jun 2018 16:08:40 +0800
Subject: [PATCH 4013/5242] MLK-18612-1 iio: adc: imx8qxp_adc: add 4 more
 input channel

commit  ecddc6548f9d6c48c5c8a7cf48cc7180cefa2c64 from
https://source.codeaurora.org/external/imx/linux-imx.git

The ADC on imx8qxp/imx8qm totally has 8 differential external
analog inputs per instance.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iio/adc/imx8qxp_adc.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/adc/imx8qxp_adc.c b/drivers/iio/adc/imx8qxp_adc.c
index 7ca509b..331e241 100644
--- a/drivers/iio/adc/imx8qxp_adc.c
+++ b/drivers/iio/adc/imx8qxp_adc.c
@@ -203,6 +203,10 @@ struct imx8qxp_adc {
 	IMX8QXP_ADC_CHAN(1),
 	IMX8QXP_ADC_CHAN(2),
 	IMX8QXP_ADC_CHAN(3),
+	IMX8QXP_ADC_CHAN(4),
+	IMX8QXP_ADC_CHAN(5),
+	IMX8QXP_ADC_CHAN(6),
+	IMX8QXP_ADC_CHAN(7),
 };
 
 static void imx8qxp_adc_feature_prepare(struct imx8qxp_adc *adc)
@@ -403,7 +407,7 @@ static int imx8qxp_adc_read_raw(struct iio_dev *indio_dev,
 		mutex_lock(&indio_dev->mlock);
 		reinit_completion(&adc->completion);
 
-		channel = chan->channel & 0x03;
+		channel = chan->channel & 0x07;
 		adc->channel_id = channel;
 		adc->cmd_id = 1;
 		adc->trigger_id = 0;
-- 
1.7.9.5

